src/Orleans.Core.Abstractions/IDs/GrainId.cs

```diff
@@ -15,6 +15,8 @@ namespace Orleans.Runtime
     [StructLayout(LayoutKind.Auto)]
     public readonly struct GrainId : IEquatable<GrainId>, IComparable<GrainId>, ISerializable
     {
+        private static readonly char[] SegmentSeparator = new[] { '/' };
```

Comment:
Everything will clash. I don't see a reason not to use this. Anything will need to be escaped if it's going to be used in a URL, etc, especially considering the components themselves currently don't have any restriction on what they can hold

Reply:
I think you understood what i was trying to say backwards.

Currently, this code when parsing will stop at the first slash, whether it was the one ToString inserted or not.

If a user included the slash in a GrainType, then the provided code will incorrectly parse the result. To avoid that would require escaping the GrainType in the ToString method, and unescaping in Parse, or (alternatively) forbidding forward slash from being part of GrainType.

Reply:
Agreed, I'm thinking of adding input sanitation soon, but not just yet. Feel free to PR them if you'd like